utils/node_local_test.py

```diff
@@ -3715,6 +3716,12 @@ def test_pil4dfs(self):
         file3 = join(path, 'file3')
         self.server.run_daos_client_cmd_pil4dfs(['touch', file3])
 
+        # cat a filename where a directory in the path is a file, should fail.  For now this does
+        # work but leaks memory.
+        # nop_file = join(file3, 'new_file which will not exist...')
+        # rc = self.server.run_daos_client_cmd_pil4dfs(['cat', nop_file], check=False)
+        # assert rc.returncode == 1, rc
+
```

Comment:
For this one I propose landing this PR with the test added-but-disabled and then you can look at the memory leak.

Reply:
I created a ticket for the memory leak issue. I will enable this test after the issue is fixed.

Reply:
https://daosio.atlassian.net/browse/DAOS-13481